ballerina/commons.bal

```diff
@@ -16,6 +16,47 @@
 
 import ballerina/time;
 
+# Represents the task service.
+public type Service distinct service object {
+  isolated function onTrigger() returns error?;
+};
+
+# Worker count for the global scheduler
+public configurable int globalSchedulerWorkerCount = 5;
+
+# Waiting time for the global scheduler
+public configurable time:Seconds globalSchedulerWaitingTime = 5;
+
+# Listener configuration.
+# 
+# + schedule - The schedule configuration for the listener
+public type ListenerConfiguration record {
+  OneTimeConfiguration|RecurringConfiguration schedule;
+};
+
+# Recurring schedule configuration.
+# 
+# + interval - The duration of the trigger (in seconds), which is used to run the job frequently
+# + maxCount - The maximum number of trigger counts
+# + startTime - The trigger start time in Ballerina `time:Civil`. If it is not provided, a trigger will
+#               start immediately
+# + endTime - The trigger end time in Ballerina `time:Civil`
+# + taskPolicy - The policy, which is used to handle the error and will be waiting during the trigger time
+public type RecurringConfiguration record {|
+  decimal interval;
+  int maxCount = -1;
+  time:Civil startTime?;
+  time:Civil endTime?;
+  TaskPolicy taskPolicy = {};
+|};
+
+# One-time schedule configuration.
+# 
+# + triggerTime - The specific time in Ballerina `time:Civil` to trigger only one time
+public type OneTimeConfiguration record {|
```

Comment:
Any reason to introduce a record with one field ? IMO we could have had following in the listener configuration:

```ballerina
public type ListenerConfiguration record {
  time:Civil|RecurringConfiguration schedule;
};
```

We could have mentioned the usage of `time:Civil` in the doc comment itself.

Reply:
If there are no other options (even if future) then +1 for this
Adding @daneshk

Reply:
Decided to use `RecurringConfiguration` config to handle both cases. The record name is changed to `TriggerConfiguration`

```ballerina
public type ListenerConfiguration record {
  TriggerConfiguration trigger;
};
```